src/Components/Aspire.Orleans.Client/Aspire.Orleans.Client.csproj

```diff
@@ -0,0 +1,29 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net8.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <Compile Include="..\Aspire.Orleans.Server\Shared\ConnectionSettings.cs" Link="Shared\ConnectionSettings.cs" />
+    <Compile Include="..\Aspire.Orleans.Server\Shared\OrleansServerSettingConstants.cs" Link="Shared\OrleansServerSettingConstants.cs" />
+    <Compile Include="..\Aspire.Orleans.Server\Shared\OrleansServerSettings.cs" Link="Shared\OrleansServerSettings.cs" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.Orleans.Clustering.AzureStorage" />
+    <PackageReference Include="Microsoft.Orleans.Client" />
+    <PackageReference Include="Microsoft.Extensions.Hosting.Abstractions" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\Aspire.Azure.Data.Tables\Aspire.Azure.Data.Tables.csproj" />
```

Comment:
I assume Orleans isn't Azure-specific. Do we want our Orleans Aspire component to be dependent on Azure?

Reply:
As stated in the first post:

> For the moment, it will support only Azure table for the [membership](https://learn.microsoft.com/en-us/dotnet/orleans/implementation/cluster-management), but we plan to add support for other membership table implementations in the future.

Reply:
That doesn't answer the question. As we add support for more implementations in the future, are we going to add them to this library? Or are we going to create separate packages for the other implementations.

Reply:
The idea is that we will have only one package compatible with all other implementations: based on the resource name we whould be able to detect which implementation is needed, and call the configuration specific via a known method passing something like `IConfiguration`. If that makes sense?

Reply:
Directly referencing the azure nuget is only for the current preview.

Reply:
>  and call the configuration specific via a known method passing something like IConfiguration

That kind of sounds like you intend to do this in a late-bound fashion? If not, can you explain a bit more how this will work when the Azure reference is removed?

Reply:
> That kind of sounds like you intend to do this in a late-bound fashion?

That's right: the current idea is to create a list of providers supported by the host process and match them with config. When a type is not supported by the host, we will throw during startup and tell the developer to add the required nuget package. The thought was to use the Orleans source-gen to embed the set of known providers into the host asm to avoid the need for asm/type scanning. Each provider would be enhanced with a static method used to configure the provider which would be given the host builder and the relevant details (`IConfigurationSection`, name).

The Aspire App Host package currently has a list of well-known providers, but we are considering how to support unknown providers. Most likely, it will be via a method to specify the provider's ID (chosen by the provider author) and the `IResourceBuilder<IResourceWithConnectionString>` instance so that the relevant details can be extracted.

Overall, the setup is attempting to give developers a one-liner to add Aspire support while retaining the ability to further customize the configuration via code if they need to.

An alternative approach is to add Aspire support to every provider package directly (~30 official ones + many community/contrib) and require the developer to call something like `.AddAzureTableGrainStorage(name)` for each, effectively mirroring what they specify in the app host. We're hoping to avoid that duplication by making this fully config driven, however, hence the current approach.